apps/site/src/app/portal/Portal.tsx

```diff
@@ -1,21 +1,62 @@
-import api from "@/lib/utils/api";
+"use client";
 
-async function getIdentity(): Promise<string> {
-	const res = await api.get<string>("/user/me");
-	return res.data;
+import Head from "next/head";
+import { useRouter } from "next/navigation";
+import { useEffect } from "react";
+
+import VerticalTimeline from "./VerticalTimeline";
+import Message from "./Message";
+import { Identity } from "@/lib/utils/getUserIdentity";
+
+import styles from "./Portal.module.scss";
+
+export const enum PortalStatus {
+	pending = "PENDING_REVIEW",
+	reviewed = "REVIEWED",
+	accepted = "ACCEPTED",
+	rejected = "REJECTED",
+	waitlisted = "WAITLISTED",
+	confirmed = "CONFIRMED",
+}
+
+interface PortalProps {
+	identity: Identity;
 }
 
-async function Portal() {
-	const identity = await getIdentity();
+function Portal({ identity }: PortalProps) {
+	const router = useRouter();
+	const status = identity.status;
+
+	useEffect(() => {
+		if (status === null) {
+			router.push("/apply");
+		}
+	}, [status, router]);
+
+	if (status === null) {
+		return null;
+	}
+
```

Comment:
Nitpick: We can use [`redirect`](https://nextjs.org/docs/app/api-reference/functions/redirect) instead of `useRouter` to maintain SSR.

Reply:
useEffect requires the component to be a client component. I will still switch over to using redirect